src/m365/spo/commands/listitem/listitem-batch-remove.ts

```diff
@@ -109,11 +109,13 @@ class SpoListItemBatchRemoveCommand extends SpoCommand {
           const fileContent = fs.readFileSync(args.options.filePath, 'utf-8');
           const jsonContent: any[] = formatting.parseCsvToJson(fileContent);
 
-          if (!jsonContent[0].hasOwnProperty('ID')) {
+          const idRow = Object.keys(jsonContent[0]).find(key => key.toLowerCase() === 'id');
+
+          if (!idRow) {
             return `The file does not contain the required header row with the column name 'ID'.`;
           }
 
-          const nonNumbers = jsonContent.filter(element => isNaN(Number(element['ID'].toString().trim()))).map(element => element['ID']);
+          const nonNumbers = jsonContent.filter(element => isNaN(Number(element[idRow].toString().trim()))).map(element => element[idRow]);
```

Comment:
In the `validation` util we have methods to check if it's a valid integer. I think `isValidPositiveInteger` can come in handy here. We also have a function to validate an array of integers.